xarray/core/dataarray.py

```diff
@@ -2295,19 +2293,27 @@ def from_dict(cls, d: dict) -> "DataArray":
         return obj
 
     @classmethod
-    def from_series(cls, series: pd.Series) -> "DataArray":
+    def from_series(cls, series: pd.Series, sparse: bool = False) -> "DataArray":
         """Convert a pandas.Series into an xarray.DataArray.
 
         If the series's index is a MultiIndex, it will be expanded into a
         tensor product of one-dimensional coordinates (filling in missing
         values with NaN). Thus this operation should be the inverse of the
         `to_series` method.
+
+        If sparse=True, creates a sparse array instead of a dense NumPy array.
+        Requires the pydata/sparse package.
+
+        See also
+        --------
+        xarray.Dataset.from_dataframe
         """
-        # TODO: add a 'name' parameter
-        name = series.name
-        df = pd.DataFrame({name: series})
-        ds = Dataset.from_dataframe(df)
-        return ds[name]
+        temp_name = "__temporary_name"
+        df = pd.DataFrame({temp_name: series})
+        ds = Dataset.from_dataframe(df, sparse=sparse)
+        result = cast(DataArray, ds[temp_name])
```

Comment:
I think I can get rid of the cast by changing ``Dataset.__getitem__``. Let's merge this one first though.

Reply:
... or not. Unlike ``@functools.single_dispatch``, ``@typing.overload`` does not allow for a more specific specific signature followed by a more generic one :(

```python

    @overload
    def __getitem__(self, key: Hashable) -> "DataArray":
        ...

    @overload
    def __getitem__(self, key: Any) -> "Dataset":
        ...

    def __getitem__(self, key):
```
mypy complains:
```
xarray/core/dataset.py:1222: error: Overloaded function signature 2 will never be matched: signature 1's parameter type(s) are the same or broader
```